src/Microsoft.AspNet.Mvc.Core/Formatters/JsonInputFormatter.cs

```diff
@@ -13,31 +14,27 @@ namespace Microsoft.AspNet.Mvc
 {
     public class JsonInputFormatter : InputFormatter
     {
-        private const int DefaultMaxDepth = 32;
-        private JsonSerializerSettings _jsonSerializerSettings;
+        private JsonSerializerSettings _serializerSettings;
 
         public JsonInputFormatter()
+            : this(SerializerSettingsHelper.CreateSerializerSettings())
         {
-            SupportedEncodings.Add(Encodings.UTF8EncodingWithoutBOM);
-            SupportedEncodings.Add(Encodings.UTF16EncodingLittleEndian);
-
-            SupportedMediaTypes.Add(MediaTypeHeaderValue.Parse("application/json"));
-            SupportedMediaTypes.Add(MediaTypeHeaderValue.Parse("text/json"));
+        }
 
-            _jsonSerializerSettings = new JsonSerializerSettings
+        public JsonInputFormatter(JsonSerializerSettings serializerSettings)
+        {
+            if (serializerSettings == null)
             {
-                MissingMemberHandling = MissingMemberHandling.Ignore,
+                throw new ArgumentNullException(nameof(serializerSettings));
+            }
```

Comment:
Add `[NotNull]` to the parameter and leave out this `null` check.  `[NotNull]` doesn't do anything at the moment but will eventually.